src/sinks/datadog/mod.rs

```diff
@@ -78,35 +73,48 @@ pub struct DatadogCommonConfig {
     pub acknowledgements: AcknowledgementsConfig,
 }
 
-impl Default for DatadogCommonConfig {
-    fn default() -> Self {
+impl DatadogCommonConfig {
```

Comment:
After this change, we now have two configs that we have to reason about within the internals of each sink. 

To consolidate and to prevent accidental misuse, how do you feel about the following proposal?

1. Make all fields private on `DatadogCommonConfig` and rename it something like `LocalDatadogCommonConfig`
```
#[configurable_component]
pub struct LocalDatadogCommonConfig {
    endpoint: Option<String>,
    site: Option<String>,
    default_api_key: Option<SensitiveString>,
    tls: Option<TlsEnableableConfig>,
    acknowledgements: AcknowledgementsConfig,
}
```
2. Create a new struct `DatadogCommonConfig`. Note that `site` and `default_api_key` are no longer optional and the fields are public. The relevant methods defined for the old `DatadogCommonConfig` will be moved to this struct.
```
pub struct DatadogCommonConfig {
    pub endpoint: Option<String>,
    pub site: String,
    pub default_api_key: SensitiveString,
    pub tls: Option<TlsEnableableConfig>,
    pub acknowledgements: AcknowledgementsConfig,
}
```
3. Create a new `LocalDatadogCommonConfig::with_globals(...) -> Result<DatadogCommonConfig>` method that merges the local and global values. Here, we will return an error if `site` or `default_api_key` are undefined, maintaining the existing config requirements.

4. Within the sinks themselves, we replace the usage of the old `DatadogCommonConfig` with the new version.

This structure would prevent the accidental misuse of the config options and avoid us having to pass around, validate, and reason about two config objects within the sinks themselves.